watertap/tools/parameter_sweep/tests/test_parameter_sweep.py

```diff
@@ -1030,11 +1024,14 @@ def test_parameter_sweep_recover(self, model, tmp_path):
                             ]
                         ),
                     },
+                    # removed upper bound tests, as in pracice
+                    # user should not be chanaing upper and lower bounds
+                    # during reinitilization and current frame work
+                    # this bounds get updated based on existing model in each worker
```

Comment:
I agree that the bounds should not change during reinitialization.

Does this test still certify that the model *was* re-initialized? That's why this was written this way, so I could be sure that the re-initialize function was actually called.

Reply:
I think we should restore this aspect of the test to its previous behavior so we *know* that the initialization function was actually called.

Reply:
I plan to completely rework the way ps tool handles init states, and will also need to redo these tests. in its current form, the test don't really make sense as you would never try to solve an uninitialized model, and re-init just supposed to run a standard initialize sequence. So we should really just test that initi function worked, and pull out the build/init sequence into its own function to remove duplicity. I plan to update the logic a bit as well and will try adding tests that test the new implementation.